### PR TITLE
Fix: add user to mailing list on signup

### PR DESCRIPTION
**Pull request by haddadzineddine**
## Problem

When a new user has registered, they are not added to the mailing list.

## Solution

To add users to the mailing list, I utilized the sympa mailing service. Additionally, I developed a generic solution that allows you to easily switch to a different service with minimal modifications to the code.

## Changes Made

- Create a sympa mailer to add registered users to the mailing list.
- Create two routes: one for subscribing and one for unsubscribing.
- Add two buttons to the account page: one for subscribing and one for unsubscribing.

## Testing

To run tests, use the following command: `HOSTNAME=stageportal.lirmm.fr rspec spec/mailers`

## Types of changes

- [x] Bug fix

### CODE REVIEW

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]
You added these two config variables $ANNOUNCE_LIST_SERVIC and $ANNOUNCE_SERVICE_HOST add it to the [bioportal_config_env.rb.sample](https://github.com/haddadzineddine/bioportal_web_ui/blob/fix/add-to-our-mailing-list/config/bioportal_config_env.rb.sample)

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]
You did not fix this 
image
make a  better UI for this

File: config/bioportal_config_env.rb.sample [hunk 1]
```diff
@@ -50,6 +50,12 @@ $REST_URL = "http://example.org:8080/"
 #$ANNOTATOR_URL = "http://services.stageportal.lirmm.fr/annotator"
 $ANNOTATOR_URL = "#{$REST_URL}/annotator"
 
+# Announcements mailman mailing list REQUEST address, EX: list-request@lists.example.org
+# NOTE: You must use the REQUEST address for the mailing list. ONLY WORKS WITH MAILMAN LISTS.
```
Comment (syphax-bouazzouni) [inline config/bioportal_config_env.rb.sample:54]:
```
# Announcements mailman mailing list REQUEST address, EX: list-request@lists.example.org
# NOTE: You must use the REQUEST address for the mailing list. ONLY WORKS WITH MAILMAN LISTS.
```
Why do you say `mailman` ?  In our case, we use `sympa`

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

**Comment by syphax-bouazzouni** [comment]
@jonquet  can you test if this works, with our mailing list? I deployed it to Stageportal

**Comment by jonquet** [comment]
@syphax-bouazzouni StagePortal has been setup with which parameters ?
$ANNOUNCE_LIST_SERVICE ||= "SERVICE_EXAMPLE"
$ANNOUNCE_SERVICE_HOST ||= "service@test.com"
$ANNOUNCE_LIST||= "users-list@test"  

I dont't see the changes in our private gite repo.

**Comment by syphax-bouazzouni** [comment]
> @syphax-bouazzouni StagePortal has been setup with which parameters ? $ANNOUNCE_LIST_SERVICE ||= "SERVICE_EXAMPLE" $ANNOUNCE_SERVICE_HOST ||= "[service@test.com](mailto:service@test.com)" $ANNOUNCE_LIST||= "users-list@test"
> 
> I dont't see the changes in our private gite repo.

See https://gite.lirmm.fr/jonquetlab/bioportal-configs/-/commit/e5a89668c7d336cdbbfd030f35b0a5e7bb3b1a07

**Comment by jonquet** [comment]
Ok. Thanks.  
- Creation of a new user (with all fieds filled) generates an error "Something went wrong" after clicking the Create button. 
- Unsuscription of an already existing user does not seems to work => I have unsuscribe my account jonquet@lirmm but double check its still on the mailing list. Maybe the mailing list is not setup to accept subscription/unsubscription by email (it might need somthing) I will check... in the meantime can we CC the support mailing list of the portal so we are sure the email was sent and the support people are informed. 

Quick comment on the flow and text messages : 

- You have un_subscribe successfully => You have successfully unsubscribed from our user mailing list: $ANNOUNCE_LIST|
- Before the button in the UI, put a sentence explaining what we are talking about.  Something like : "Subscribe to the *portalName* user mailing list : [BOUTON O/N] qui affiche l'état actuel de la subscription

image

**Comment by haddadzineddine** [comment]
TODO :

- [ ] test new user creation with annouce list enabled
- [ ]  send (un)subscribtion mail to our support mailing list in paralle
- [ ] update success subscribe message to add annouce list in the end
- [ ] update the ui , create button [ O/N ]

**Comment by syphax-bouazzouni** [comment]
Hi @haddadzineddine , don't forget to write meaningful commit messages.
image

Do a rebase https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message

File: app/views/users/show.html.haml [hunk 2]
```diff
@@ -132,9 +132,36 @@
...
+            %label.switch
+              %input{type: "checkbox", id: "unsubscribe", data: {url: '/toggle'}}
+              %span Unsubscribe from the #{$ANNOUNCE_LIST} user mailing list
+              %span.slider.round
+
+            
+            :javascript
+              document.getElementById('subscribe').addEventListener('change', function() {
+                if (this.checked) {
+                  window.location.href = "/users/subscribe/" + "#{@user.username}"
```
Comment (syphax-bouazzouni) [inline app/views/users/show.html.haml:156]: Don't forget to protect  this two endpoints to be accessed only the owner and admins

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

**Review by syphax-bouazzouni — APPROVED** [review]

**Pull request merged**